docs: Update "Create reports" doc to clarify custom time ranges and add information about drafts (#69015)
grafana/grafana: docs/sources/dashboards/create-reports/index.md

   - **Send test email:** To verify that the configuration works as expected. You can choose to send this email to the recipients configured for the report, or to a different set of email addresses only used for testing.
1. Preview and save the report.

### Save as draft
{{% admonition type="note" %}}
Available in [Grafana Enterprise]({{< relref "../../introduction/grafana-enterprise/" >}}) version 9.1.0 and later and [Grafana Cloud Pro and Advanced](/docs/grafana-cloud/).
{{% /admonition %}}

> You can save a report as a draft at any point during the report creation or update process. You can save a report as a draft even if it is missing required fields, and the report will not be sent according to its schedule while it is a draft.

Reviewer: ```suggestion
You can save a report as a draft at any point during the report creation or update process. You can save a report as a draft even if it's missing required fields. Also, the report won't be sent according to its schedule while it's a draft.
```
